app/src/main/java/de/xikolo/utils/extensions/TextViewExtensions.kt

```diff
@@ -1,162 +1,90 @@
-
-    companion object {
-        val TAG: String = BypassGlideImageGetter::class.java.simpleName
-    }
-
-    private val textViewWeakReference: WeakReference<TextView> = WeakReference(textView)
-
-    private var maxWidth = -1
-
-    override fun getDrawable(source: String): Drawable {
-
-        val result = BitmapDrawablePlaceHolder()
-
-        object : AsyncTask<Void, Void, Bitmap>() {
-
-            override fun doInBackground(vararg meh: Void): Bitmap? {
-                var uri = Uri.parse(source)
-                if (uri.isRelative) {
-                    uri = Uri.Builder()
-                        .scheme("https")
-                        .authority(Config.HOST)
-                        .path(uri.path)
-                        .query(uri.query)
-                        .build()
-                }
-                try {
-                    return glideRequests
-                        .asBitmap()
-                        .load(uri)
-                        .centerCrop()
-                        .noPlaceholders()
-                        .dontAnimate()
-                        .submit()
-                        .get()
-                } catch (e: Exception) {
-                    Log.e(TAG, e.message, e)
-                    return null
+        fun getAbsoluteUrl(url: String): String {
+            try {
+                val uri = URI(url)
+                if (!uri.isAbsolute) {
+                    return URL(URL(Config.HOST_URL), url).toString()
                 }
-
+            } catch (e: URISyntaxException) {
+                return url
+            } catch (e: MalformedURLException) {
+                return url
             }
+            return url
+        }
 
-            override fun onPostExecute(bitmap: Bitmap) {
-                val textView = textViewWeakReference.get() ?: return
-                try {
-                    if (maxWidth == -1) {
-                        val horizontalPadding = textView.paddingLeft + textView.paddingRight
-                        maxWidth = textView.measuredWidth - horizontalPadding
-                        if (maxWidth == 0) {
-                            maxWidth = Integer.MAX_VALUE
-                        }
+        val glide = Glide.with(context)
+        val formatter = Markwon.builder(context)
+            .usePlugin(
+                GlideImagesPlugin.create(object : GlideImagesPlugin.GlideStore {
+                    override fun cancel(target: Target<*>) {
+                        glide.clear(target)
                     }
 
-                    val drawable = BitmapDrawable(textView.resources, bitmap)
-                    val aspectRatio = 1.0 * drawable.intrinsicWidth / drawable.intrinsicHeight
-
-                    // real image width in pixel scaled based on screen density
-                    val scaledWidth = Math.round(drawable.intrinsicWidth * Resources.getSystem().displayMetrics.density)
```

Comment:
I remember that I implemented that to keep the image proportions similar to the web. Is that handled well by the lib out of the box?

E.g., images with a large width like in this [course description](https://open.hpi.de/courses/prototype2019) or [this](https://open.hpi.de/courses/international-teams2019).

Reply:
I observed the behavior, that it does not expand past the container width if it is larger but keeps its original size (is not stretched) when it is smaller than container width.
<img src="https://user-images.githubusercontent.com/26904189/90630997-47d30a80-e222-11ea-9905-70bf65552e41.png" width="250px"/>

Reply:
Looks good. 👍